--- qt-2.3.10/src/kernel/qwindowsystem_qws.h.orig	2005-09-19 17:50:39.000000000 +0200
+++ qt-2.3.10/src/kernel/qwindowsystem_qws.h	2005-09-19 17:50:45.000000000 +0200
@@ -53,6 +53,8 @@
 class QWSRegionManager;
 class QGfx;
 class QWSServerData;
+class QWSInputMethod;
+class QWSGestureMethod;
 
 class QWSInternalWindowInfo
 {
--- qt-2.3.10/src/tools/qvaluestack.h.orig	2005-01-24 03:00:47.000000000 +1300
+++ qt-2.3.10/src/tools/qvaluestack.h	2007-12-22 21:21:21.000000000 +1300
@@ -54,7 +54,7 @@
     {
 	T elem( this->last() );
 	if ( !this->isEmpty() )
-	    remove( this->fromLast() );
+	    this->remove( this->fromLast() );
 	return elem;
     }
     T& top() { return this->last(); }
