--- ./arch/alpha/Kconfig.org	2002-10-16 17:09:12.000000000 +0200
+++ ./arch/alpha/Kconfig	2002-10-16 17:12:17.000000000 +0200
@@ -791,10 +791,10 @@
 
 endmenu
 
-#if [ "$CONFIG_PCI" = "y" ]; then
+if PCI
 source "drivers/message/fusion/Kconfig"
+endif
 
-#fi
 source "net/Kconfig"
 
 
@@ -1046,5 +1046,3 @@
 
 source "lib/Kconfig"
 
-source "Kconfig"
-
--- ./arch/arm/Kconfig.org	2002-10-16 17:09:13.000000000 +0200
+++ ./arch/arm/Kconfig	2002-10-16 17:12:17.000000000 +0200
@@ -876,10 +876,10 @@
 
 source "drivers/parport/Kconfig"
 
-#if [ "$CONFIG_ALIGNMENT_TRAP" = "y" ]; then
+if ALIGNMENT_TRAP
 source "drivers/mtd/Kconfig"
+endif
 
-#else
 source "drivers/pnp/Kconfig"
 
 source "drivers/block/Kconfig"
@@ -1233,5 +1233,3 @@
 
 source "lib/Kconfig"
 
-source "Kconfig"
-
--- ./arch/cris/Kconfig.org	2002-10-16 17:09:14.000000000 +0200
+++ ./arch/cris/Kconfig	2002-10-16 17:12:17.000000000 +0200
@@ -749,5 +749,3 @@
 
 source "lib/Kconfig"
 
-source "Kconfig"
-
--- ./arch/i386/Kconfig.org	2002-10-16 17:09:15.000000000 +0200
+++ ./arch/i386/Kconfig	2002-10-16 17:12:17.000000000 +0200
@@ -1430,7 +1430,8 @@
 	  Read the file <file:Documentation/svga.txt> for more information
 	  about the Video mode selection support. If unsure, say N.
 
-#if [ "$CONFIG_EXPERIMENTAL" = "y" ]; then
+if EXPERIMENTAL
+
 config MDA_CONSOLE
 	tristate "MDA text console (dual-headed) (EXPERIMENTAL)"
 	---help---
@@ -1449,7 +1450,8 @@
 
 source "drivers/video/Kconfig"
 
-#fi
+endif
+
 endmenu
 
 
@@ -1625,5 +1627,3 @@
 	bool
 	default y
 
-source "Kconfig"
-
--- ./arch/ia64/Kconfig.org	2002-10-16 17:09:16.000000000 +0200
+++ ./arch/ia64/Kconfig	2002-10-16 17:12:17.000000000 +0200
@@ -120,43 +120,10 @@
 
 choice
 	prompt "Kernel page size"
-	depends on ITANIUM
-	default IA64_PAGE_SIZE_16KB_1
-
-config IA64_PAGE_SIZE_4KB_1
-	bool "4KB"
-
-config IA64_PAGE_SIZE_8KB_1
-	bool "8KB"
-
-config IA64_PAGE_SIZE_16KB_1
-	bool "16KB"
-
-endchoice
-
-choice
-	prompt "Kernel page size"
-	depends on !ITANIUM
-	default IA64_PAGE_SIZE_16KB_2
-
-config IA64_PAGE_SIZE_4KB_2
-	bool "4KB"
-
-config IA64_PAGE_SIZE_8KB_2
-	bool "8KB"
-
-config IA64_PAGE_SIZE_16KB_2
-	bool "16KB"
-
-config IA64_PAGE_SIZE_64KB
-	bool "64KB"
-
-endchoice
+	default IA64_PAGE_SIZE_16KB
 
 config IA64_PAGE_SIZE_4KB
-	bool
-	depends on IA64_PAGE_SIZE_4KB_1 || IA64_PAGE_SIZE_4KB_2
-	default y
+	bool "4KB"
 	---help---
 	  This lets you select the page size of the kernel.  For best IA-64
 	  performance, a page size of 8KB or 16KB is recommended.  For best
@@ -173,14 +140,16 @@
 	  If you don't know what to do, choose 8KB.
 
 config IA64_PAGE_SIZE_8KB
-	bool
-	depends on IA64_PAGE_SIZE_8KB_1 || IA64_PAGE_SIZE_8KB_2
-	default y
+	bool "8KB"
 
 config IA64_PAGE_SIZE_16KB
-	bool
-	depends on IA64_PAGE_SIZE_16KB_1 || IA64_PAGE_SIZE_16KB_2
-	default y
+	bool "16KB"
+
+config IA64_PAGE_SIZE_64KB
+	depends on !ITANIUM
+	bool "64KB"
+
+endchoice
 
 config ACPI
 	bool
@@ -403,87 +372,33 @@
 
 choice
 	prompt "IA-64 Huge TLB Page Size"
-	depends on HUGETLB_PAGE && MCKINLEY
-	default HUGETLB_PAGE_SIZE_16MB_1
+	depends on HUGETLB_PAGE
+	default HUGETLB_PAGE_SIZE_16MB
 
 config HUGETLB_PAGE_SIZE_4GB
+	depends on MCKINLEY
 	bool "4GB"
 
-config HUGETLB_PAGE_SIZE_256MB_1
-	bool "256MB"
-
-config HUGETLB_PAGE_SIZE_64MB_1
-	bool "64MB"
-
-config HUGETLB_PAGE_SIZE_16MB_1
-	bool "16MB"
-
-config HUGETLB_PAGE_SIZE_4MB_1
-	bool "4MB"
-
-config HUGETLB_PAGE_SIZE_1MB_1
-	bool "1MB"
-
-config HUGETLB_PAGE_SIZE_256KB_1
-	bool "256KB"
-
-endchoice
-
-choice
-	prompt "IA-64 Huge TLB Page Size"
-	depends on HUGETLB_PAGE && !MCKINLEY
-	default HUGETLB_PAGE_SIZE_16MB_2
-
-config HUGETLB_PAGE_SIZE_256MB_2
+config HUGETLB_PAGE_SIZE_256MB
 	bool "256MB"
 
-config HUGETLB_PAGE_SIZE_64MB_2
+config HUGETLB_PAGE_SIZE_64MB
 	bool "64MB"
 
-config HUGETLB_PAGE_SIZE_16MB_2
+config HUGETLB_PAGE_SIZE_16MB
 	bool "16MB"
 
-config HUGETLB_PAGE_SIZE_4MB_2
+config HUGETLB_PAGE_SIZE_4MB
 	bool "4MB"
 
-config HUGETLB_PAGE_SIZE_1MB_2
+config HUGETLB_PAGE_SIZE_1MB
 	bool "1MB"
 
-config HUGETLB_PAGE_SIZE_256KB_2
+config HUGETLB_PAGE_SIZE_256KB
 	bool "256KB"
 
 endchoice
 
-config HUGETLB_PAGE_SIZE_256MB
-	bool
-	depends on HUGETLB_PAGE_SIZE_256MB_1 || HUGETLB_PAGE_SIZE_256MB_2
-	default y
-
-config HUGETLB_PAGE_SIZE_64MB
-	bool
-	depends on HUGETLB_PAGE_SIZE_64MB_1 || HUGETLB_PAGE_SIZE_64MB_2
-	default y
-
-config HUGETLB_PAGE_SIZE_16MB
-	bool
-	depends on HUGETLB_PAGE_SIZE_16MB_1 || HUGETLB_PAGE_SIZE_16MB_2
-	default y
-
-config HUGETLB_PAGE_SIZE_4MB
-	bool
-	depends on HUGETLB_PAGE_SIZE_4MB_1 || HUGETLB_PAGE_SIZE_4MB_2
-	default y
-
-config HUGETLB_PAGE_SIZE_1MB
-	bool
-	depends on HUGETLB_PAGE_SIZE_1MB_1 || HUGETLB_PAGE_SIZE_1MB_2
-	default y
-
-config HUGETLB_PAGE_SIZE_256KB
-	bool
-	depends on HUGETLB_PAGE_SIZE_256KB_1 || HUGETLB_PAGE_SIZE_256KB_2
-	default y
-
 config SMP
 	bool "SMP support"
 	---help---
@@ -608,7 +523,8 @@
 	  you have use for it; the module is called binfmt_misc.o. If you
 	  don't know what to answer at this point, say Y.
 
-#if [ "$CONFIG_IA64_HP_SIM" = "n" ]; then
+if !IA64_HP_SIM
+
 source "drivers/acpi/Kconfig"
 
 config PCI
@@ -650,10 +566,12 @@
 
 source "drivers/parport/Kconfig"
 
-#fi # !HP_SIM
+endif
+
 endmenu
 
-#if [ "$CONFIG_IA64_HP_SIM" = "n" ]; then
+if !IA64_HP_SIM
+
 source "drivers/mtd/Kconfig"
 
 source "drivers/pnp/Kconfig"
@@ -721,7 +639,8 @@
 
 endmenu
 
-#fi
+endif
+
 
 menu "SCSI support"
 
@@ -751,7 +670,8 @@
 
 source "net/Kconfig"
 
-#if [ "$CONFIG_IA64_HP_SIM" = "n" ]; then
+
+if !IA64_HP_SIM
 
 menu "Network device support"
 	depends on NET
@@ -841,7 +761,8 @@
 #source drivers/misc/Config.in
 source "drivers/media/Kconfig"
 
-#else # HP_SIM
+endif
+
 
 menu "Block devices"
 	depends on IA64_HP_SIM
@@ -863,10 +784,10 @@
 
 endmenu
 
-# HP_SIM
 source "fs/Kconfig"
 
-#if [ "$CONFIG_IA64_HP_SIM" = "n" ]; then
+if !IA64_HP_SIM
+
 
 menu "Console drivers"
 	depends on VT
@@ -936,7 +857,8 @@
 
 source "net/bluetooth/Kconfig"
 
-#fi # !HP_SIM
+endif
+
 source "arch/ia64/hp/sim/Kconfig"
 
 
@@ -1058,5 +980,3 @@
 
 source "security/Kconfig"
 
-source "Kconfig"
-
--- ./arch/m68k/Kconfig.org	2002-10-16 17:09:18.000000000 +0200
+++ ./arch/m68k/Kconfig	2002-10-16 17:12:17.000000000 +0200
@@ -634,10 +634,10 @@
 
 source "drivers/zorro/Kconfig"
 
-#if [ "$CONFIG_Q40" = "y" ]; then
+if Q40
 source "drivers/pnp/Kconfig"
+endif
 
-#fi
 endmenu
 
 source "drivers/mtd/Kconfig"
@@ -646,10 +646,10 @@
 
 source "drivers/md/Kconfig"
 
-#if [ "$CONFIG_MAC" = "y" ]; then
+if MAC
 source "drivers/input/Kconfig"
+endif
 
-#fi
 
 menu "ATA/ATAPI/MFM/RLL device support"
 
@@ -2382,5 +2382,3 @@
 
 source "lib/Kconfig"
 
-source "Kconfig"
-
--- ./arch/mips/Kconfig.org	2002-10-16 17:09:20.000000000 +0200
+++ ./arch/mips/Kconfig	2002-10-16 17:12:17.000000000 +0200
@@ -810,10 +810,11 @@
 #    fi
 endmenu
 
-#if [ "$CONFIG_ISA" = "y" ]; then
+
+if ISA
 source "drivers/pnp/Kconfig"
+endif
 
-#fi
 source "drivers/mtd/Kconfig"
 
 source "drivers/parport/Kconfig"
@@ -903,11 +904,11 @@
 
 endmenu
 
-#if [ "$CONFIG_DECSTATION" != "y" -a \
-#     "$CONFIG_SGI_IP22" != "y" ]; then
+
+if !DECSTATION" && !SGI_IP22
 source "drivers/message/i2o/Kconfig"
+endif
 
-#fi
 source "net/Kconfig"
 
 
@@ -1200,7 +1201,8 @@
 
 	  Say Y.
 
-#if [ "$CONFIG_EXPERIMENTAL" = "y" ]; then
+if EXPERIMENTAL
+
 config MDA_CONSOLE
 	tristate "MDA text console (dual-headed) (EXPERIMENTAL)"
 	---help---
@@ -1219,7 +1221,8 @@
 
 source "drivers/video/Kconfig"
 
-#fi
+endif
+
 endmenu
 
 
@@ -1333,5 +1336,3 @@
 
 source "lib/Kconfig"
 
-source "Kconfig"
-
--- ./arch/mips64/Kconfig.org	2002-10-16 17:09:22.000000000 +0200
+++ ./arch/mips64/Kconfig	2002-10-16 17:12:17.000000000 +0200
@@ -735,5 +735,3 @@
 
 source "lib/Kconfig"
 
-source "Kconfig"
-
--- ./arch/parisc/Kconfig.org	2002-10-16 17:09:22.000000000 +0200
+++ ./arch/parisc/Kconfig	2002-10-16 17:12:17.000000000 +0200
@@ -759,5 +759,3 @@
 
 source "lib/Kconfig"
 
-source "Kconfig"
-
--- ./arch/ppc/Kconfig.org	2002-10-16 17:09:23.000000000 +0200
+++ ./arch/ppc/Kconfig	2002-10-16 17:12:17.000000000 +0200
@@ -1826,5 +1826,3 @@
 
 source "security/Kconfig"
 
-source "Kconfig"
-
--- ./arch/ppc64/Kconfig.org	2002-10-16 17:09:24.000000000 +0200
+++ ./arch/ppc64/Kconfig	2002-10-16 17:12:17.000000000 +0200
@@ -525,5 +525,3 @@
 
 source "lib/Kconfig"
 
-source "Kconfig"
-
--- ./arch/s390/Kconfig.org	2002-10-16 17:09:25.000000000 +0200
+++ ./arch/s390/Kconfig	2002-10-16 17:12:17.000000000 +0200
@@ -318,5 +318,3 @@
 
 source "lib/Kconfig"
 
-source "Kconfig"
-
--- ./arch/s390x/Kconfig.org	2002-10-16 17:09:25.000000000 +0200
+++ ./arch/s390x/Kconfig	2002-10-16 17:12:17.000000000 +0200
@@ -326,5 +326,3 @@
 
 source "lib/Kconfig"
 
-source "Kconfig"
-
--- ./arch/sh/Kconfig.org	2002-10-16 17:09:25.000000000 +0200
+++ ./arch/sh/Kconfig	2002-10-16 17:12:17.000000000 +0200
@@ -1214,7 +1214,8 @@
 	  Read the file <file:Documentation/svga.txt> for more information
 	  about the Video mode selection support. If unsure, say N.
 
-#if [ "$CONFIG_EXPERIMENTAL" = "y" ]; then
+if EXPERIMENTAL
+
 config MDA_CONSOLE
 	tristate "MDA text console (dual-headed) (EXPERIMENTAL)"
 	---help---
@@ -1233,7 +1234,8 @@
 
 source "drivers/video/Kconfig"
 
-#fi
+endif
+
 endmenu
 
 
@@ -1319,5 +1321,3 @@
 
 source "lib/Kconfig"
 
-source "Kconfig"
-
--- ./arch/sparc/Kconfig.org	2002-10-16 17:09:28.000000000 +0200
+++ ./arch/sparc/Kconfig	2002-10-16 17:12:17.000000000 +0200
@@ -238,7 +238,8 @@
 	  a kernel compiled with this option will run only on Sun4.
 	  (And the current version will probably work only on sun4/330.)
 
-#if [ "$CONFIG_SUN4" != "y" ]; then
+if !SUN4
+
 config PCI
 	bool "Support for PCI and PS/2 keyboard/mouse"
 	help
@@ -254,7 +255,8 @@
 
 source "drivers/pci/Kconfig"
 
-#else
+endif
+
 config SUN_OPENPROMFS
 	tristate "Openprom tree appears in /proc/openprom"
 	help
@@ -429,10 +431,10 @@
 
 source "drivers/serial/Kconfig"
 
-#if [ "$CONFIG_SUN4" != "y" ]; then
+if !SUN4
 source "drivers/sbus/char/Kconfig"
+endif
 
-#fi
 
 menu "Block devices"
 
@@ -575,7 +577,7 @@
 endmenu
 
 # Don't frighten a common SBus user
-#if [ "$CONFIG_PCI" = "y" ]; then
+if PCI
 
 menu "ATA/ATAPI/MFM/RLL support"
 
@@ -629,7 +631,8 @@
 
 endmenu
 
-#else
+endif
+
 source "drivers/isdn/Kconfig"
 
 
@@ -1419,5 +1422,3 @@
 
 source "lib/Kconfig"
 
-source "Kconfig"
-
--- ./arch/sparc64/Kconfig.org	2002-10-16 17:09:28.000000000 +0200
+++ ./arch/sparc64/Kconfig	2002-10-16 17:12:17.000000000 +0200
@@ -978,12 +978,58 @@
 	  whenever you want). If you want to compile it as a module, say M
 	  here and read <file:Documentation/modules.txt>.
 
-#if [ "$CONFIG_PCI" != "n" ]; then
+
+choice
+	prompt "Adaptec AIC7xxx support"
+	optional
+	depends on SCSI && PCI
+
 source "drivers/scsi/aic7xxx/Kconfig"
 
+config SCSI_AIC7XXX_OLD
+	tristate "Old driver"
+	---help---
+	  WARNING This driver is an older aic7xxx driver and is no longer
+	  under active development.  Adaptec, Inc. is writing a new driver to
+	  take the place of this one, and it is recommended that whenever
+	  possible, people should use the new Adaptec written driver instead
+	  of this one.  This driver will eventually be phased out entirely.
+
+	  This is support for the various aic7xxx based Adaptec SCSI
+	  controllers. These include the 274x EISA cards; 284x VLB cards;
+	  2902, 2910, 293x, 294x, 394x, 3985 and several other PCI and
+	  motherboard based SCSI controllers from Adaptec. It does not support
+	  the AAA-13x RAID controllers from Adaptec, nor will it likely ever
+	  support them. It does not support the 2920 cards from Adaptec that
+	  use the Future Domain SCSI controller chip. For those cards, you
+	  need the "Future Domain 16xx SCSI support" driver.
+
+	  In general, if the controller is based on an Adaptec SCSI controller
+	  chip from the aic777x series or the aic78xx series, this driver
+	  should work. The only exception is the 7810 which is specifically
+	  not supported (that's the RAID controller chip on the AAA-13x
+	  cards).
+
+	  Note that the AHA2920 SCSI host adapter is *not* supported by this
+	  driver; choose "Future Domain 16xx SCSI support" instead if you have
+	  one of those.
+
+	  Information on the configuration options for this controller can be
+	  found by checking the help file for each of the available
+	  configuration options. You should read
+	  <file:drivers/scsi/aic7xxx_old/README.aic7xxx> at a minimum before
+	  contacting the maintainer with any questions.  The SCSI-HOWTO,
+	  available from <http://www.linuxdoc.org/docs.html#howto>, can also
+	  be of great help.
+
+	  If you want to compile this driver as a module ( = code which can be
+	  inserted in and removed from the running kernel whenever you want),
+	  say M here and read <file:Documentation/modules.txt>.  The module
+	  will be called aic7xxx_old.o.
+
 config AIC7XXX_OLD_TCQ_ON_BY_DEFAULT
 	bool "Enable Tagged Command Queueing (TCQ) by default"
-	depends on SCSI_AIC7XXX!=y && SCSI_AIC7XXX_OLD
+	depends on SCSI_AIC7XXX_OLD
 	---help---
 	  This option causes the aic7xxx driver to attempt to use Tagged
 	  Command Queueing (TCQ) on all devices that claim to support it.
@@ -1019,7 +1065,7 @@
 
 config AIC7XXX_OLD_CMDS_PER_DEVICE
 	int "Maximum number of TCQ commands per device"
-	depends on SCSI_AIC7XXX!=y && SCSI_AIC7XXX_OLD
+	depends on SCSI_AIC7XXX_OLD
 	default "8"
 	---help---
 	  Specify the number of commands you would like to allocate per SCSI
@@ -1043,7 +1089,7 @@
 
 config AIC7XXX_OLD_PROC_STATS
 	bool "Collect statistics to report in /proc"
-	depends on SCSI_AIC7XXX!=y && SCSI_AIC7XXX_OLD
+	depends on SCSI_AIC7XXX_OLD
 	---help---
 	  This option tells the driver to keep track of how many commands have
 	  been sent to each particular device and report that information to
@@ -1056,6 +1102,8 @@
 
 	  If unsure, say N.
 
+endchoice
+
 config SCSI_SYM53C8XX_2
 	tristate "SYM53C8XX Version 2 SCSI support"
 	depends on PCI && SCSI
@@ -1346,10 +1394,10 @@
 
 source "drivers/fc4/Kconfig"
 
-#if [ "$CONFIG_PCI" = "y" ]; then
+if PCI
 source "drivers/message/fusion/Kconfig"
+endif
 
-#fi
 source "drivers/ieee1394/Kconfig"
 
 source "net/Kconfig"
@@ -1651,5 +1699,3 @@
 
 source "lib/Kconfig"
 
-source "Kconfig"
-
--- ./arch/x86_64/Kconfig.org	2002-10-16 17:09:29.000000000 +0200
+++ ./arch/x86_64/Kconfig	2002-10-16 17:12:17.000000000 +0200
@@ -592,7 +592,8 @@
 	  Read the file <file:Documentation/svga.txt> for more information
 	  about the Video mode selection support. If unsure, say N.
 
-#if [ "$CONFIG_EXPERIMENTAL" = "y" ]; then
+if EXPERIMENTAL
+
 config MDA_CONSOLE
 	tristate "MDA text console (dual-headed) (EXPERIMENTAL)"
 	---help---
@@ -611,7 +612,8 @@
 
 source "drivers/video/Kconfig"
 
-#fi
+endif
+
 endmenu
 
 
@@ -713,5 +715,3 @@
 
 source "lib/Kconfig"
 
-source "Kconfig"
-
--- ./arch/um/Kconfig.org	2002-10-16 17:09:29.000000000 +0200
+++ ./arch/um/Kconfig	2002-10-16 17:12:17.000000000 +0200
@@ -163,5 +163,3 @@
 
 endmenu
 
-source "Kconfig"
-
--- ./drivers/isdn/eicon/Kconfig.org	2002-10-16 17:09:15.000000000 +0200
+++ ./drivers/isdn/eicon/Kconfig	2002-10-16 17:12:17.000000000 +0200
@@ -11,12 +11,12 @@
 	  <file:Documentation/isdn/README.eicon> for more information.
 
 choice
-	prompt "change me!"
+	prompt "Eicon active card support"
 	optional
 	depends on ISDN_DRV_EICON && ISDN
 
 config ISDN_DRV_EICON_DIVAS
-	tristate "Build Eicon driver type standalone"
+	tristate "Eicon driver"
 	depends on PCI
 	help
 	  Enable this option if you want the eicon driver as standalone
@@ -26,16 +26,14 @@
 	  here.
 
 config ISDN_DRV_EICON_OLD
-	tristate "Legacy Eicon driver"
+	tristate "Legacy driver"
 	help
 	  Say Y here to use your Eicon active ISDN card with ISDN4Linux
 	  isdn module.
 
-endchoice
-
 config ISDN_DRV_EICON_PCI
 	bool "Eicon PCI DIVA Server BRI/PRI/4BRI support"
-	depends on ISDN_DRV_EICON_DIVAS!=y && ISDN_DRV_EICON_OLD && PCI
+	depends on ISDN_DRV_EICON_OLD && PCI
 	help
 	  Say Y here if you have an Eicon Diva Server (BRI/PRI/4BRI) ISDN
 	  card.  Please read <file:Documentation/isdn/README.eicon> for more
@@ -43,7 +41,7 @@
 
 config ISDN_DRV_EICON_ISA
 	bool "Eicon S,SX,SCOM,Quadro,S2M support"
-	depends on ISDN_DRV_EICON_DIVAS!=y && ISDN_DRV_EICON_OLD
+	depends on ISDN_DRV_EICON_OLD
 	help
 	  Say Y here if you have an old-type Eicon active ISDN card. In order
 	  to use this card, additional firmware is necessary, which has to be
@@ -51,3 +49,5 @@
 	  the latest isdn4k-utils package. Please read the file
 	  <file:Documentation/isdn/README.eicon> for more information.
 
+endchoice
+
--- ./drivers/scsi/aic7xxx/Kconfig.org	2002-10-16 17:09:15.000000000 +0200
+++ ./drivers/scsi/aic7xxx/Kconfig	2002-10-16 17:12:17.000000000 +0200
@@ -1,51 +1,5 @@
-choice
-	prompt "change me!"
-	optional
-	depends on SCSI
-
-config SCSI_AIC7XXX_OLD
-	tristate "Old Adaptec AIC7xxx support"
-	---help---
-	  WARNING This driver is an older aic7xxx driver and is no longer
-	  under active development.  Adaptec, Inc. is writing a new driver to
-	  take the place of this one, and it is recommended that whenever
-	  possible, people should use the new Adaptec written driver instead
-	  of this one.  This driver will eventually be phased out entirely.
-
-	  This is support for the various aic7xxx based Adaptec SCSI
-	  controllers. These include the 274x EISA cards; 284x VLB cards;
-	  2902, 2910, 293x, 294x, 394x, 3985 and several other PCI and
-	  motherboard based SCSI controllers from Adaptec. It does not support
-	  the AAA-13x RAID controllers from Adaptec, nor will it likely ever
-	  support them. It does not support the 2920 cards from Adaptec that
-	  use the Future Domain SCSI controller chip. For those cards, you
-	  need the "Future Domain 16xx SCSI support" driver.
-
-	  In general, if the controller is based on an Adaptec SCSI controller
-	  chip from the aic777x series or the aic78xx series, this driver
-	  should work. The only exception is the 7810 which is specifically
-	  not supported (that's the RAID controller chip on the AAA-13x
-	  cards).
-
-	  Note that the AHA2920 SCSI host adapter is *not* supported by this
-	  driver; choose "Future Domain 16xx SCSI support" instead if you have
-	  one of those.
-
-	  Information on the configuration options for this controller can be
-	  found by checking the help file for each of the available
-	  configuration options. You should read
-	  <file:drivers/scsi/aic7xxx_old/README.aic7xxx> at a minimum before
-	  contacting the maintainer with any questions.  The SCSI-HOWTO,
-	  available from <http://www.linuxdoc.org/docs.html#howto>, can also
-	  be of great help.
-
-	  If you want to compile this driver as a module ( = code which can be
-	  inserted in and removed from the running kernel whenever you want),
-	  say M here and read <file:Documentation/modules.txt>.  The module
-	  will be called aic7xxx_old.o.
-
 config SCSI_AIC7XXX
-	tristate "Adaptec AIC7xxx support"
+	tristate "New driver"
 	help
 	  This driver supports all of Adaptec's PCI based SCSI controllers
 	  (not the hardware RAID controllers though) as well as the aic7770
@@ -54,11 +8,10 @@
 	  intended to replace the previous aic7xxx driver maintained by Doug
 	  Ledford since Doug is no longer maintaining that driver.
 
-endchoice
 
 config AIC7XXX_CMDS_PER_DEVICE
 	int "Maximum number of TCQ commands per device"
-	depends on SCSI_AIC7XXX_OLD!=y && SCSI_AIC7XXX
+	depends on SCSI_AIC7XXX
 	default "253"
 	---help---
 	  Specify the number of commands you would like to allocate per SCSI
@@ -77,7 +30,7 @@
 
 config AIC7XXX_RESET_DELAY_MS
 	int "Initial bus reset delay in milli-seconds"
-	depends on SCSI_AIC7XXX_OLD!=y && SCSI_AIC7XXX
+	depends on SCSI_AIC7XXX
 	default "15000"
 	help
 	  The number of milliseconds to delay after an initial bus reset.
@@ -88,7 +41,7 @@
 
 config AIC7XXX_BUILD_FIRMWARE
 	bool "Build Adapter Firmware with Kernel Build"
-	depends on SCSI_AIC7XXX_OLD!=y && SCSI_AIC7XXX
+	depends on SCSI_AIC7XXX
 	help
 	  This option should only be enabled if you are modifying the firmware
 	  source to the aic7xxx driver and wish to have the generated firmware
--- ./drivers/scsi/Kconfig.org	2002-10-16 17:09:15.000000000 +0200
+++ ./drivers/scsi/Kconfig	2002-10-16 17:12:17.000000000 +0200
@@ -311,8 +311,57 @@
 	tristate "Adaptec AACRAID support (EXPERIMENTAL)"
 	depends on EXPERIMENTAL && SCSI && PCI
 
+choice
+	prompt "Adaptec AIC7xxx support"
+	optional
+	depends on SCSI
+
 source "drivers/scsi/aic7xxx/Kconfig"
 
+config SCSI_AIC7XXX_OLD
+	tristate "Old driver"
+	help
+	  WARNING This driver is an older aic7xxx driver and is no longer
+	  under active development.  Adaptec, Inc. is writing a new driver to
+	  take the place of this one, and it is recommended that whenever
+	  possible, people should use the new Adaptec written driver instead
+	  of this one.  This driver will eventually be phased out entirely.
+
+	  This is support for the various aic7xxx based Adaptec SCSI
+	  controllers. These include the 274x EISA cards; 284x VLB cards;
+	  2902, 2910, 293x, 294x, 394x, 3985 and several other PCI and
+	  motherboard based SCSI controllers from Adaptec. It does not support
+	  the AAA-13x RAID controllers from Adaptec, nor will it likely ever
+	  support them. It does not support the 2920 cards from Adaptec that
+	  use the Future Domain SCSI controller chip. For those cards, you
+	  need the "Future Domain 16xx SCSI support" driver.
+
+	  In general, if the controller is based on an Adaptec SCSI controller
+	  chip from the aic777x series or the aic78xx series, this driver
+	  should work. The only exception is the 7810 which is specifically
+	  not supported (that's the RAID controller chip on the AAA-13x
+	  cards).
+
+	  Note that the AHA2920 SCSI host adapter is *not* supported by this
+	  driver; choose "Future Domain 16xx SCSI support" instead if you have
+	  one of those.
+
+	  Information on the configuration options for this controller can be
+	  found by checking the help file for each of the available
+	  configuration options. You should read
+	  <file:drivers/scsi/aic7xxx_old/README.aic7xxx> at a minimum before
+	  contacting the maintainer with any questions.  The SCSI-HOWTO,
+	  available from <http://www.linuxdoc.org/docs.html#howto>, can also
+	  be of great help.
+
+	  If you want to compile this driver as a module ( = code which can be
+	  inserted in and removed from the running kernel whenever you want),
+	  say M here and read <file:Documentation/modules.txt>.  The module
+	  will be called aic7xxx_old.o.
+
+endchoice
+
+
 # All the I2O code and drivers do not seem to be 64bit safe.
 config SCSI_DPT_I2O
 	tristate "Adaptec I2O RAID support "
--- ./sound/oss/Kconfig.org	2002-10-16 17:09:15.000000000 +0200
+++ ./sound/oss/Kconfig	2002-10-16 17:12:17.000000000 +0200
@@ -1023,12 +1023,12 @@
 	  DSP 16 card.
 
 choice
-	prompt "change me!"
+	prompt "Audio Excel DSP 16"
 	optional
 	depends on SOUND_AEDSP16
 
 config AEDSP16_MSS
-	bool "Audio Excel DSP 16 (MSS emulation)"
+	bool "MSS emulation"
 	depends on SOUND_MSS
 	help
 	  Answer Y if you want your audio card to emulate Microsoft Sound
@@ -1036,7 +1036,7 @@
 	  and say N to "Audio Excel DSP 16 (SBPro emulation)".
 
 config AEDSP16_SBPRO
-	bool "Audio Excel DSP 16 (SBPro emulation)"
+	bool "SBPro emulation"
 	depends on SOUND_SB
 	help
 	  Answer Y if you want your audio card to emulate Sound Blaster Pro.
