--- src.orig/widgets/qstatusbar.cpp	2002-09-02 16:10:07.000000000 +0200
+++ src/widgets/qstatusbar.cpp	2002-09-02 16:10:41.000000000 +0200
@@ -442,13 +442,15 @@
 */
 void QStatusBar::resizeEvent( QResizeEvent * e )
 {
-    QStatusBarPrivate::SBItem* item;    
+#if 0
+    QStatusBarPrivate::SBItem* item;
     for ( item = d->items.first(); item; item = d->items.next() )
 	item->w->setMinimumWidth( 30 );
 
     int mw = d->box->totalMinimumSize().width() - 30;
     for ( item = d->items.first(); item; item = d->items.next() )
 	item->w->setMaximumWidth( width() - mw );
+#endif
 
     QWidget::resizeEvent( e );
 }
