--- src.orig/kernel/qapplication.cpp	Mon Mar 18 02:11:25 2002
+++ src/kernel/qapplication.cpp	Fri Aug 30 04:28:34 2002
@@ -930,10 +930,10 @@
 void QApplication::setStyle( QStyle *style )
 {
     QStyle* old = app_style;
-    app_style = style;
 
     if ( startingUp() ) {
 	delete old;
+	app_style = style;
 	return;
     }
 
@@ -954,6 +954,8 @@
 	old->unPolish( qApp );
     }
 
+    app_style = style;
+ 
     // take care of possible palette requirements of certain gui
     // styles. Do it before polishing the application since the style
     // might call QApplication::setStyle() itself
