--- src/kernel/qapplication_qws.cpp	2002-12-08 21:51:08.000000000 +0100
+++ src/kernel/qapplication_qws.cpp.orig	2002-12-08 18:06:22.000000000 +0100
@@ -3278,7 +3278,7 @@
 	if ( qApp->inPopupMode() ) {			// in popup mode
 	    QWidget *popup = qApp->activePopupWidget();
 	    if ( popup != this ) {
-		if ( testWFlags(WType_Popup) && rect().contains(pos) && !popup->rect().contains(popup-> mapFromGlobal(mapToGlobal(pos))) )
+		if ( testWFlags(WType_Popup) && rect().contains(pos) )
 		    popup = this;
 		else				// send to last popup
 		    pos = popup->mapFromGlobal( globalPos );
