--- src/widgets/qlistview.cpp.orig	2002-11-23 23:31:19.000000000 +0100
+++ src/widgets/qlistview.cpp	2002-11-24 00:08:51.000000000 +0100
@@ -4968,9 +4968,9 @@
 	l = l->childItem ? l->childItem : l->siblingItem;
 
     if ( l && l->height() )
-	s.setHeight( s.height() + 10 * l->height() );
-    else
-	s.setHeight( s.height() + 140 );
+	s.setHeight( s.height() + 4 /*10*/ * l->height() );
+    else                      // ^v much too big for handhelds
+	s.setHeight( s.height() + 30 /*140*/ );
 
     if ( s.width() > s.height() * 3 )
 	s.setHeight( s.width() / 3 );
