--- src/Makefile.in.orig	2002-12-06 21:00:04.000000000 +0100
+++ src/Makefile.in	2002-12-06 21:02:37.000000000 +0100
@@ -6,7 +6,7 @@
 CXXFLAGS=	$(SYSCONF_CXXFLAGS_QT) $(SYSCONF_CXXFLAGS_X11) $(SYSCONF_CXXFLAGS) $(SYSCONF_CXXFLAGS_LIB) $(QT_CXXFLAGS_OPT)
 CC	=	$(SYSCONF_CC) $(QT_C_MT)
 CFLAGS	=	$(SYSCONF_CFLAGS) $(SYSCONF_CFLAGS_LIB) $(QT_CFLAGS_OPT)
-INCPATH =	-I3rdparty/libpng -I3rdparty/zlib -I3rdparty/freetype/src -I3rdparty/freetype/include -I3rdparty/freetype/builds/unix -I3rdparty/kernel -I3rdparty/tools
+INCPATH =	-I3rdparty/freetype/src -I3rdparty/freetype/include -I3rdparty/freetype/builds/unix -I3rdparty/kernel -I3rdparty/tools
 LFLAGS	=	$(SYSCONF_LFLAGS) $(QT_LFLAGS_MT)
 LIBS	=	$(SUBLIBS)  $(SYSCONF_LIBS_QTLIB)
 MOC	=	$(SYSCONF_MOC)
@@ -24,104 +24,8 @@
 ####### Files
 
 HEADERS =	
-SOURCES =	3rdparty/libpng/png.c \
-		3rdparty/libpng/pngerror.c \
-		3rdparty/libpng/pngget.c \
-		3rdparty/libpng/pngmem.c \
-		3rdparty/libpng/pngpread.c \
-		3rdparty/libpng/pngread.c \
-		3rdparty/libpng/pngrio.c \
-		3rdparty/libpng/pngrtran.c \
-		3rdparty/libpng/pngrutil.c \
-		3rdparty/libpng/pngset.c \
-		3rdparty/libpng/pngtrans.c \
-		3rdparty/libpng/pngwio.c \
-		3rdparty/libpng/pngwrite.c \
-		3rdparty/libpng/pngwtran.c \
-		3rdparty/libpng/pngwutil.c \
-		3rdparty/zlib/adler32.c \
-		3rdparty/zlib/compress.c \
-		3rdparty/zlib/crc32.c \
-		3rdparty/zlib/deflate.c \
-		3rdparty/zlib/gzio.c \
-		3rdparty/zlib/infblock.c \
-		3rdparty/zlib/infcodes.c \
-		3rdparty/zlib/inffast.c \
-		3rdparty/zlib/inflate.c \
-		3rdparty/zlib/inftrees.c \
-		3rdparty/zlib/infutil.c \
-		3rdparty/zlib/trees.c \
-		3rdparty/zlib/uncompr.c \
-		3rdparty/zlib/zutil.c \
-		3rdparty/freetype/builds/unix/ftsystem.c \
-		3rdparty/freetype/src/base/ftdebug.c \
-		3rdparty/freetype/src/base/ftinit.c \
-		3rdparty/freetype/src/base/ftbase.c \
-		3rdparty/freetype/src/base/ftglyph.c \
-		3rdparty/freetype/src/base/ftmm.c \
-		3rdparty/freetype/src/base/ftbbox.c \
-		3rdparty/freetype/src/autohint/autohint.c \
-		3rdparty/freetype/src/cache/ftcache.c \
-		3rdparty/freetype/src/cff/cff.c \
-		3rdparty/freetype/src/cid/type1cid.c \
-		3rdparty/freetype/src/psaux/psaux.c \
-		3rdparty/freetype/src/psnames/psmodule.c \
-		3rdparty/freetype/src/raster/raster.c \
-		3rdparty/freetype/src/sfnt/sfnt.c \
-		3rdparty/freetype/src/smooth/smooth.c \
-		3rdparty/freetype/src/truetype/truetype.c \
-		3rdparty/freetype/src/type1/type1.c \
-		3rdparty/freetype/src/winfonts/winfnt.c \
-		allmoc.cpp
-OBJECTS =	3rdparty/libpng/png.o \
-		3rdparty/libpng/pngerror.o \
-		3rdparty/libpng/pngget.o \
-		3rdparty/libpng/pngmem.o \
-		3rdparty/libpng/pngpread.o \
-		3rdparty/libpng/pngread.o \
-		3rdparty/libpng/pngrio.o \
-		3rdparty/libpng/pngrtran.o \
-		3rdparty/libpng/pngrutil.o \
-		3rdparty/libpng/pngset.o \
-		3rdparty/libpng/pngtrans.o \
-		3rdparty/libpng/pngwio.o \
-		3rdparty/libpng/pngwrite.o \
-		3rdparty/libpng/pngwtran.o \
-		3rdparty/libpng/pngwutil.o \
-		3rdparty/zlib/adler32.o \
-		3rdparty/zlib/compress.o \
-		3rdparty/zlib/crc32.o \
-		3rdparty/zlib/deflate.o \
-		3rdparty/zlib/gzio.o \
-		3rdparty/zlib/infblock.o \
-		3rdparty/zlib/infcodes.o \
-		3rdparty/zlib/inffast.o \
-		3rdparty/zlib/inflate.o \
-		3rdparty/zlib/inftrees.o \
-		3rdparty/zlib/infutil.o \
-		3rdparty/zlib/trees.o \
-		3rdparty/zlib/uncompr.o \
-		3rdparty/zlib/zutil.o \
-		3rdparty/freetype/builds/unix/ftsystem.o \
-		3rdparty/freetype/src/base/ftdebug.o \
-		3rdparty/freetype/src/base/ftinit.o \
-		3rdparty/freetype/src/base/ftbase.o \
-		3rdparty/freetype/src/base/ftglyph.o \
-		3rdparty/freetype/src/base/ftmm.o \
-		3rdparty/freetype/src/base/ftbbox.o \
-		3rdparty/freetype/src/autohint/autohint.o \
-		3rdparty/freetype/src/cache/ftcache.o \
-		3rdparty/freetype/src/cff/cff.o \
-		3rdparty/freetype/src/cid/type1cid.o \
-		3rdparty/freetype/src/psaux/psaux.o \
-		3rdparty/freetype/src/psnames/psmodule.o \
-		3rdparty/freetype/src/raster/raster.o \
-		3rdparty/freetype/src/sfnt/sfnt.o \
-		3rdparty/freetype/src/smooth/smooth.o \
-		3rdparty/freetype/src/truetype/truetype.o \
-		3rdparty/freetype/src/type1/type1.o \
-		3rdparty/freetype/src/winfonts/winfnt.o \
-		allmoc.o $(QT_MODULE_OBJ)
+SOURCES =	allmoc.cpp
+OBJECTS =	allmoc.o $(QT_MODULE_OBJ)
 INTERFACES =    
 UICDECLS =      
 UICIMPLS =      
