--- src/kernel/qgfxraster_qws.cpp.orig	2002-12-08 05:02:49.000000000 +0100
+++ src/kernel/qgfxraster_qws.cpp	2002-12-08 05:04:09.000000000 +0100
@@ -4058,7 +4058,7 @@
     setAlphaType(IgnoreAlpha);
     if ( w <= 0 || h <= 0 ) return;
     GFX_START(QRect(rx+xoffs, ry+yoffs, w+1, h+1))
-#ifdef QWS_EXPERIMENTAL_FASTPATH
+#if 0 // def QWS_EXPERIMENTAL_FASTPATH    !! this is crashing HancomWord on OZ !!
     // ### fix for 8bpp
     // This seems to be reliable now, at least for 16bpp
 
