--- qt-2.3.4-beta5/src/Makefile.in.orig	2003-01-30 19:37:44.000000000 -0800
+++ qt-2.3.4-beta5/src/Makefile.in	2003-02-12 16:02:09.000000000 -0800
@@ -6,7 +6,7 @@
 CXXFLAGS=	$(SYSCONF_CXXFLAGS_QT) $(SYSCONF_CXXFLAGS_X11) $(SYSCONF_CXXFLAGS) $(SYSCONF_CXXFLAGS_LIB) -DQT_NO_IMAGEIO_MNG $(QT_CXXFLAGS_OPT)
 CC	=	$(SYSCONF_CC) $(QT_C_MT)
 CFLAGS	=	$(SYSCONF_CFLAGS) $(SYSCONF_CFLAGS_LIB) -DQT_NO_IMAGEIO_MNG $(QT_CFLAGS_OPT)
-INCPATH =	-I3rdparty/freetype/src -I3rdparty/freetype/include -I3rdparty/freetype/builds/unix -I3rdparty/kernel -I3rdparty/tools
+INCPATH =	-I3rdparty/kernel -I3rdparty/tools
 LFLAGS	=	$(SYSCONF_LFLAGS) $(QT_LFLAGS_MT)
 LIBS	=	$(SUBLIBS)  $(SYSCONF_LIBS_QTLIB)
 MOC	=	$(SYSCONF_MOC)
@@ -44,26 +44,7 @@
 		3rdparty/freetype/src/type1/type1.c \
 		3rdparty/freetype/src/winfonts/winfnt.c \
 		allmoc.cpp
-OBJECTS =	3rdparty/freetype/builds/unix/ftsystem.o \
-		3rdparty/freetype/src/base/ftdebug.o \
-		3rdparty/freetype/src/base/ftinit.o \
-		3rdparty/freetype/src/base/ftbase.o \
-		3rdparty/freetype/src/base/ftglyph.o \
-		3rdparty/freetype/src/base/ftmm.o \
-		3rdparty/freetype/src/base/ftbbox.o \
-		3rdparty/freetype/src/autohint/autohint.o \
-		3rdparty/freetype/src/cache/ftcache.o \
-		3rdparty/freetype/src/cff/cff.o \
-		3rdparty/freetype/src/cid/type1cid.o \
-		3rdparty/freetype/src/psaux/psaux.o \
-		3rdparty/freetype/src/psnames/psmodule.o \
-		3rdparty/freetype/src/raster/raster.o \
-		3rdparty/freetype/src/sfnt/sfnt.o \
-		3rdparty/freetype/src/smooth/smooth.o \
-		3rdparty/freetype/src/truetype/truetype.o \
-		3rdparty/freetype/src/type1/type1.o \
-		3rdparty/freetype/src/winfonts/winfnt.o \
-		allmoc.o $(QT_MODULE_OBJ)
+OBJECTS =	allmoc.o $(QT_MODULE_OBJ)
 INTERFACES =    
 UICDECLS =      
 UICIMPLS =      
